pkg/ambex/transforms.go

```diff
@@ -135,10 +142,20 @@ func V3ListenerToRdsListener(lnr *v3listener.Listener) (*v3listener.Listener, []
 					if rc.Name == "" {
 						// Generate a unique name for the RouteConfiguration that we can use to
 						// correlate the listener to the RDS record. We use the listener name plus
-						// an index because there can be more than one route configuration
+						// filter_chain_match hash because there can be more than one route configuration
 						// associated with a given listener.
-						rc.Name = fmt.Sprintf("%s-routeconfig-%d", l.Name, len(routes))
+						filterChainMatch, _ := json.Marshal(fc.GetFilterChainMatch())
```

Comment:
@kflynn - Is there any downside to using the `json` package vs `protojson`. Based on the docs it says it can produce a different output https://pkg.go.dev/google.golang.org/protobuf/encoding/protojson. Which probably in this scenario isn't important since it just being used for a hash. Assuming between go-versions it doesn't change.

again, I don't think there is an issue here but just asking for curiosity sake.

Reply:
Ugh, I missed this. I don't believe that this is a problem: the `protojson` package is mostly about supporting protobuf-typed-stuff, and this isn't.

Reply:
No worries :)